src/App.tsx

```diff
@@ -9,42 +9,41 @@ import RecipeList from "./containers/RecipeList";
 import RecipeDetail from "./containers/RecipeDetail";
 import AddRecipeForm from "./components/AddRecipeForm";
 
-class App extends Component {
-  constructor(props) {
-    super(props);
+interface IAppState {
+  isAddingRecipe: boolean
+  selectedRecipe: boolean
+}
 
-    this.state = {
-      isAddingRecipe: false
-    };
+class App extends Component<{}, IAppState> {
+  state: IAppState = {
+    isAddingRecipe: false,
+    selectedRecipe: false
```

Comment:
Since your constructor function just instantiated state, you can do that outside and ditch the rest of the constructor